File: frontend/src/types/analysis.ts
/**
 * 分析相关类型定义
 */

import type { MarketType } from './common'

export type AnalysisStatus = 'pending' | 'running' | 'completed' | 'failed'

export type AnalysisDepth = 1 | 2 | 3 | 4 | 5

export interface AnalysisTask {
  id: string
  user_id: string
  symbol: string
  market: MarketType
  analysis_date: string
  depth: AnalysisDepth
  status: AnalysisStatus
  progress: number
  result?: AnalysisResult
  error_message?: string
  created_at: string
  updated_at: string
}

export interface AnalysisResult {
  symbol: string
  market: MarketType
  fundamental?: FundamentalAnalysis
  technical?: TechnicalAnalysis
  sentiment?: SentimentAnalysis
  news?: NewsAnalysis
  recommendation: Recommendation
  risk_assessment: RiskAssessment
}

export interface FundamentalAnalysis {
  summary: string
  metrics: {
    pe_ratio?: number
    pb_ratio?: number
    roe?: number
    gross_margin?: number
    [key: string]: number | undefined
  }
  analysis: string
}

export interface TechnicalAnalysis {
  summary: string
  indicators: {
    [key: string]: number
  }
  signals: string[]
  analysis: string
}

export interface SentimentAnalysis {
  summary: string
  score: number
  analysis: string
}

export interface NewsAnalysis {
  summary: string
  news_items: NewsItem[]
  analysis: string
}

export interface NewsItem {
  title: string
  source: string
  publish_time: string
  url?: string
  sentiment?: number
}

export interface Recommendation {
  action: 'STRONG_BUY' | 'BUY' | 'NEUTRAL' | 'SELL' | 'STRONG_SELL'
  confidence: number
  target_price?: number
  stop_loss?: number
  reasoning: string
}

export interface RiskAssessment {
  risk_level: 'LOW' | 'MEDIUM' | 'HIGH'
  risk_factors: string[]
  mitigation_strategies: string[]
}

export interface AnalysisProgress {
  task_id: string
  progress: number
  stage: string
  message: string
  timestamp: string
}

export interface StartAnalysisRequest {
  symbol: string
  market: MarketType
  analysis_date?: string
  depth?: AnalysisDepth
}

File: backend/app/schemas/analysis.py
"""分析 Schemas"""

from datetime import datetime
from pydantic import BaseModel, Field
from uuid import UUID


class AnalysisTaskCreate(BaseModel):
    """创建分析任务"""

    symbol: str = Field(..., min_length=1, max_length=20)
    market: str = Field(..., pattern="^(CN|HK|US)$")
    analysis_date: str | None = None
    depth: int = Field(default=3, ge=1, le=5)


class AnalysisResult(BaseModel):
    """分析结果"""

    symbol: str
    market: str
    fundamental: dict | None = None
    technical: dict | None = None
    sentiment: dict | None = None
    news: dict | None = None
    recommendation: dict
    risk_assessment: dict


class AnalysisTaskResponse(BaseModel):
    """分析任务响应"""

    id: UUID
    user_id: UUID
    symbol: str
    market: str
    analysis_date: str
    depth: int
    status: str
    progress: int
    result: dict | None = None
    error_message: str | None = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class AnalysisProgress(BaseModel):
    """分析进度"""

    task_id: str
    progress: int
    stage: str
    message: str
    timestamp: datetime
